lively.ide/studio/controls/svg.cp.js

```diff
@@ -0,0 +1,233 @@
+  }
+
+  onRefresh (prop) {
+    if (prop === 'target') this.update();
+  }
+
+  confirm () {
+    if (!this.target) return;
+    let color = this.ui.fillColorInput.colorValue;
+    this.target.css({ fill: color.toString(), 'fill-opacity': color.a });
+  }
+
+  async update () {
+    // TODO: Fixme, color must not be defined via css in the svg
+    // this.ui.fillColorInput.setColor(this.target.css.fill);
+  }
+
+  deactivate () {
+    this.models.fillColorInput.closeColorPicker();
+  }
+}
+
+export class SVGBorderControlModel extends ViewModel {
+  static get properties () {
+    return {
+      target: {},
+      bindings: {
+        get () {
+          return [
+            {
+              model: 'border color input',
+              signal: 'color',
+              handler: 'confirm'
+            },
+            {
+              target: 'border width input',
+              signal: 'number',
+              handler: 'confirm'
+            },
+            {
+              model: 'border style selector',
+              signal: 'selection',
+              handler: 'confirm'
+            }
+          ];
+        }
+      }
+    };
+  }
+
+  deactivate () {
+    this.models.borderColorInput.closeColorPicker();
+  }
+
+  /**
+   * Update the current border value based on the inputs in the UI.
+   * This is invoked in response to user interactions.
+   */
+  confirm () {
+    if (!this.target) return;
+    const { borderColorInput, borderWidthInput, borderStyleSelector } = this.ui;
+    this.target.css({
+      stroke: borderColorInput.colorValue,
+      'stroke-width': borderWidthInput.number,
+      'stroke-dasharray': borderStyleSelector.selection,
+      'stroke-linecap': 'round'
+    });
+  }
+}
+
+export class SVGControlModel extends ViewModel {
+  focusOn (target) {
+    this.target = target;
+    Object.values(this.models).forEach(each => each.target = this.target);
+  }
+
+  deactivate () {
+    this.view.visible = false;
+    Object.values(this.models).forEach(each => { if (each !== this) each.deactivate(); });
+  }
+}
+
+// part(SVGFillControl).openInWorld()
+
+const SVGFillControl = component(PropertySection, {
+  defaultViewModel: SVGFillControlModel,
+  name: 'svg fill control',
+  fill: Color.transparent,
+  layout: new TilingLayout({
+    axis: 'column',
+    orderByIndex: true,
+    padding: rect(0, 10, 0, 10),
+    resizePolicies: [['h floater', {
+      height: 'fixed',
+      width: 'fill'
+    }]],
+    spacing: 10,
+    wrapSubmorphs: false
+  }),
+  extent: pt(250, 121),
+  submorphs: [
+    {
+      name: 'h floater',
+      submorphs: [
+        without('add button'), {
+          name: 'section headline',
+          textAndAttributes: ['Fill', null]
+        }]
+    },
+    add(part(ColorInput, {
+      name: 'fill color input',
+      viewModel: {
+        gradientEnabled: true,
+        colorPickerComponent: DarkColorPicker
+      },
+      extent: pt(250, 27)
+    }))
+
+  ]
+});
+
+// part(SVGBorderControl).openInWorld()
```

Comment:
what is that for? it is also in line 102 e.g.?

Reply:
This is for testing and debugging only